MapboxNavigation/RouteTableViewController.swift

```diff
@@ -46,6 +47,40 @@ class RouteTableViewController: UIViewController {
             headerView.timeRemaining.text = String.localizedStringWithFormat(NSLocalizedString("LESS_THAN", bundle: .mapboxNavigation, value: "<%@", comment: "Format string for less than; 1 = duration remaining"), dateComponentsFormatter.string(from: 61)!)
         } else {
             headerView.timeRemaining.text = dateComponentsFormatter.string(from: routeProgress.durationRemaining)
+            
+            guard let coordinates = routeProgress.route.coordinates else { return }
+            
+            // To keep this cheap, use the fraction traveled along the current route.
```

Comment:
Aren’t we considering the total of expected travel times along all the remaining segments of the route? We could recompute that every time this method is called, which you’ve studiously avoided, but one optimization would be to essentially index the segments by step so that it would be cheaper to get all the remaining segments starting from the current location. So when this method is called, we’d sum the expected travel times on all the remaining segments on the current step plus all the remaining steps.